aoc/internal/array.go

```diff
@@ -0,0 +1,45 @@
+package internal
+
+import (
+	"errors"
+)
+
+var EmptyError = errors.New("Iterable has no items.")
+
+func Min[I int64 | float64 | int](items []I) (I, error) {
```

Comment:
Look into `Ordered` constraint. You may be able to use it here. `func Min[T constraint.Ordered](items []T) (T, error)`
https://pkg.go.dev/golang.org/x/exp/constraints#Ordered

Reply:
Ahh I had a feeling something like that must exist!

Reply:
> Warning: Packages here are experimental and unreliable. Some may one day be promoted to the main repository or other subrepository, or they may be modified arbitrarily or even disappear altogether.

![sus](https://user-images.githubusercontent.com/35393936/205519616-fd5d5ef9-efc9-490c-8180-b91f66baaef9.png)

Reply:
Where is this?

Reply:
https://pkg.go.dev/golang.org/x/exp#section-readme